packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -85,13 +92,8 @@ import {
     ParserDiagnostics,
 } from '@lwc/errors';
 
-function getKeyGenerator() {
-    let count = 1;
-    return () => count++;
-}
-
 function isStyleElement(irElement: IRElement) {
-    const element = irElement.__original as parse5.AST.Default.Element;
+    const element = irElementMap.get(irElement) as parse5.AST.Default.Element;
```

Comment:
thinking if we can do this without getting the original element. We already have 'tag' property on the IRElement so we can check for 'style', not sure how critical it is to check for namespaceUri though. When we have better typeing and have a dedicated StyleElement type, then we can ensure to to take namespaceUri into accounts.